### ISSUE TITLE
js helpers not loading from the correct location

### ISSUE THREAD

**Issue by lorennorman** [cypress-io/cypress#15]
I have this directory structure:
![2015-04-07 at 2 09 pm](https://cloud.githubusercontent.com/assets/17697/7030036/b9bc760c-dd2f-11e4-83b6-a1ee9e1cbc2b.png)

...and this is my cypress.json:
```
{
  "cypress": {
    ...
    "testFolder": "spec/cypress",
    "javascripts": ["support/spec_helper.coffee"]
  }
}
```

That `javascripts` line makes it hit this URL:
`http://localhost:3000/support/spec_helper.coffee`

...which 404s. I *think* it is supposed to be hitting this one:
`http://localhost:3000/tests/support/spec_helper.coffee`

(if I change the line to `"javascripts": ["tests/support/spec_helper.coffee"]` then it works, but obviously that's not what I should be doing)

Looks like a find/replace error where you removed `tests` from one place you shouldn't have, maybe? Or didn't finish changing away from `tests` to the json-injected location.

Thanks! :sparkles:

**Comment by brian-mann** [comment]
Okay yah, `javascripts` paths are always relative to your projectRoot directory.  I can certainly see how you'd assume it to be relative to your `testFolder`.  The idea was that `javascripts` may contain other files besides support files, or files within the `testFolder` itself.

But that may be the minority scenario, and if so I'd gladly change this to be relative to `testFolder`.

Alternatively, this is all part of `configuration`, which I intend to build a GUI for which can intelligently diagnose your settings and figure out whether it can even **find** files at those paths.  Then the implementation details aren't nearly as important.

**Comment by lorennorman** [comment]
Actually, root-relative is where I started, and that would be:
`spec/cypress/support/spec_helper.coffee`
...which does not work.

Again, it only works when I pull `tests` out of the sky (there is no folder named `tests` in my project at this time.)

**Comment by brian-mann** [comment]
Okay this is fixed in `0.4.1`.  [Changelog](https://github.com/cypress-io/cypress/wiki/Changelog#041-04082015)

`javascripts` are now **correctly** assumed to be a path from the root of your project.

You'll need to change the paths in your `javascripts` array to read: `spec/cypress/support/spec_helper.coffee`

They are now preprocessed in the same way as your spec files, which means they naturally support `snockets`, `browserify`, or `requirejs`.

Previously this didn't work because but it wasn't pre-processing the file.  Therefore it was sending raw `coffeescript`.

Prepending the `tests` actually tricked Cypress into thinking it was an actual spec file and assuming it could be found from within your `testFolder` and preprocessing it.  Obviously that is super weird and not supported.  And it just worked because your support file happened to be in the `testFolder`.  

I've changed the url strategy to prevent that now.

I also realized I'm not currently supporting globbing in the `javascripts` array, but I will add that in the next few days.
